File: examfrontapp/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private  _http:HttpClient) { }
  public quizzes()
  {
    return this._http.get(`${baseUrl}/quiz/`)
  }
  public addQuiz(quiz:any)
  {
   return  this._http.post(`${baseUrl}/quiz/`,quiz);
  }
  public deleteQuiz(qid:any)
  {
    return this._http.delete(`${baseUrl}/quiz/${qid}`)
  }
  public getQuiz(qid:any)
  {
    return this._http.get(`${baseUrl}/quiz/${qid}`);
  }
  public updatequiz(quiz:any)
  {
    return this._http.put(`${baseUrl}/quiz/`,quiz)
  }
  public getQuizzesofCategory(cid:any)
  {
    return this._http.get(`${baseUrl}/quiz/category/${cid}`)
  }
  public getActiveQuizzes()
  {
    return this._http.get(`${baseUrl}/quiz/active`);
  }
  public getActiveQuizzesofCategory(cid:any)
  {
    return this._http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }
}

File: examserver (1)/examserver/src/main/java/com/example/examserver/service/QuizService.java
package com.example.examserver.service;

import com.example.examserver.model.exam.Category;
import com.example.examserver.model.exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {
    public Quiz addQuiz(Quiz quiz);
    public Quiz updateQuiz(Quiz quiz);
    public Set<Quiz> getQuizzes();
    public Quiz getQuiz(Long quizId);
    public void deleteQuiz(Long quizId);

    public List<Quiz> getQuizzesofCategory(Category category);
    public List<Quiz> getActiveQuizzes();
    public  List<Quiz> getActiveQuizzesofCategory(Category c);
}
